synapse/server_notices/resource_limits_server_notices.py

```diff
@@ -91,6 +91,15 @@ def maybe_send_server_notice_to_user(self, user_id):
         currently_blocked, ref_events = yield self._is_room_currently_blocked(room_id)
 
         try:
+            if not self._config.mau_limit_alerting:
```

Comment:
the big stack of if statements in this function is getting really hard to follow, and I'm worried that the duplicated code will get out of sync.

I think it would be clearer to make the `check_auth_blocking` call (ie, everything between lines 103 and 112 as it currently stands) conditional on `mau_limit_alerting` (leaving `is_auth_blocking = False` if `mau_limit_alerting` is false).

Reply:
@richvdh I've had a go a refactoring the whole area, I _think_ it is now easier to read, albeit still inherently fiddly.

Reply:
I guess you didn't agree with my suggestion? Fair enough. It's certainly clearer now.

Reply:
For reference https://gist.github.com/richvdh/092934e994a2443bd184a01357c98677 Agreed that proposal is neater, but less explicit about the blocking behaviour, so will not adopt.